tests/pykafka/test_balancedconsumer.py

```diff
@@ -201,6 +201,19 @@ def test_external_kazoo_client(self):
         messages = [msg for msg in consumer]
         consumer.stop()
 
+
+    def test_no_partitions(self):
+        """Ensure a consumer assigned no partitions immediately exits"""
+        consumer = self.client.topics[self.topic_name].get_balanced_consumer(
+                b'test_no_partitions',
+                zookeeper_connect=self.kafka.zookeeper,
+                auto_start=False)
+        consumer._decide_partitions = lambda p: set()
+        consumer.start()
+        time.sleep(1)
```

Comment:
Because the rebalance during `start()` runs on the same thread, it also works without this sleep. Not that it adds anything noticeable to the time our integration tests take anyway ;)